server/memstore.go

```diff
@@ -366,6 +386,40 @@ func (ms *memStore) expireMsgs() {
 	}
 }
 
+// PurgeEx will remove messages based on subject filters, sequence and number of messages to keep.
+// Will return the number of purged messages.
+func (ms *memStore) PurgeEx(subject string, sequence, keep uint64) (purged uint64, err error) {
+	if subject == _EMPTY_ || subject == ">" && keep == 0 {
+		return ms.Purge()
+	}
+	eq := compareFn(subject)
+	if ss := ms.FilteredState(1, subject); ss.Msgs > 0 {
+		if keep > 0 {
+			if keep > ss.Msgs {
+				return 0, nil
+			}
+			ss.Msgs -= keep
+		}
+		last := ss.Last
+		if sequence > 0 {
+			last = sequence + 1
```

Comment:
You have:
```
// Purge up to but not including sequence.
Sequence uint64 `json:"seq,omitempty"`
```
So I have a stream with subject "foo" and "bar", each with 5 messages. I sent request to purge "foo" with Sequence: 4, I expect this subject to have only messages 1, 2, 3 removed, so total messages left in the stream should be 2 + 5 = 7. But instead, the 5 messages from "foo" are removed.
Question is: is the comment for the option wrong or should it be that `last` is equal to `sequence-1`?

Reply:
That code changed in a subsequent commit.

Sequences are not tracked per subject, so when using sequence in this regard it is absolute to the whole stream.